src/libs/telemetry.js

```diff
@@ -5,68 +5,81 @@ import Error from '../adapters/error';
 
 const telemetry = nconf.get().telemetry;
 const system = nconf.get().system;
-const telemtryEventUrl = 'events';
+const telemetryEventUrl = 'events';
 const uniqEventList = [];
+const events = {};
 
-export default class Telemetry {
-  constructor() {}
+/*
+  This converts "/src/libs/telemetry.js" events into a map where you can use an event as follow:
+  'app_start' event will be accessible like this "Telemetry.APP_START" and its value will be the
+  original (so 'app_start').
+*/
+telemetryModule.events.forEach(event => {
+  events[event.toUpperCase()] = event;
+});
 
-  static add(event, type, source, data) {
-    if (event) {
-      if (type && source) {
-        const event_const_name = `${(type + '_' + source + '_' + event).toUpperCase()}`;
-        event = Telemetry[event_const_name];
-      }
-      return Telemetry.send(event, data);
+function add(event, type, source, data) {
+  if (event) {
+    if (type && source) {
+      const event_const_name = `${(type + '_' + source + '_' + event).toUpperCase()}`;
+      event = events[event_const_name];
     }
-    Error.send('telemetry', 'add', 'telemetry event mismatch configuration', {});
+
+    return send(event, data);
   }
 
-  static addOnce(event) {
-    if (uniqEventList.indexOf(event) === -1) {
-      uniqEventList.push(event);
-      Telemetry.add(event);
-    }
+  Error.send('telemetry', 'add', 'telemetry event mismatch configuration', {});
+}
+
+function addOnce(event) {
+  if (uniqEventList.indexOf(event) === -1) {
+    uniqEventList.push(event);
+    add(event);
   }
+}
 
-  static async send(event, extra_data) {
-    if (!telemetry.enabled) {
-      return;
-    } else if (typeof event === 'undefined') {
-      Error.send('telemetry', 'send', 'unknown event received', {});
-      return;
-    }
-    const data = { type: event };
-    if (typeof extra_data === 'object' && extra_data !== null) {
-      Object.keys(extra_data).forEach(key => {
-        data[key] = extra_data[key];
-      });
-    }
-    const telemetryUrl = `${system.baseUrl}${telemtryEventUrl}`;
-    return Ajax.post(telemetryUrl, data);
+function send(event, extra_data) {
+  const telemetryUrl = `${system.baseUrl}${telemetryEventUrl}`;
```

Comment:
It made more sense to have the early return conditions before declaring stuff that may not be used.